src/dynamodb.js

```diff
@@ -1,23 +1,43 @@
 const assert = require('assert');
+const drop = require('lodash/drop');
 const get = require('lodash/get');
 const pick = require('lodash/pick');
 const set = require('lodash/set');
+const take = require('lodash/take');
 
 const rxjs = require('rxjs');
 const { from, EMPTY, of, combineLatest, BehaviorSubject } = require('rxjs');
-const { bufferCount, concatAll, scan, switchMap, concatMap, map } = require('rxjs/operators');
+const { bufferCount, concatAll, scan, switchMap, concatMap, map, takeWhile } = require('rxjs/operators');
 
 const DELETE = 'DeleteRequest';
 const PUT = 'PutRequest';
 
 const DYNAMODB_BATCH_SIZE = 25;
+const END_MARKER = Symbol('End of mock query');
 const END_OF_PAGES = Symbol('END_OF_PAGES');
 const NO_KEY_KNOWN = Symbol('NO_KEY_KNOWN');
 
 exports.delete = function deleteItems (keys = [ 'id' ]) {
   return map((item) => ({ ...item, key: pick(item.item, keys), operation: DELETE }));
 };
 
+exports.mockItemQueryStream = (items, itemRequests) => {
```

Comment:
Do you foresee future iterations that involve exposing more mocks? I wonder if a clearer separation from 'app code' functions would help avoid confusion from outside consumers of this project. Maybe a separate mocks file that is exposed as delta.mocks.<something>, for example... but could be structured otherwise.

Reply:
Ya, I wasn't really sure what the best way to organize this is. This particular function seems like something that will be well utilized, but I'm not sure if we will want to add more or not. We could also wait and see and re-org when we see. Do you feel strongly one way or the other?

Reply:
Not really - just wanted to see what your plans were. I'm happy to approve for now and see how things evolve.